src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -82,7 +81,7 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
             resetScrollToCoords={{ x: 0, y: 0 }}
             enableOnAndroid
             keyboardShouldPersistTaps='always'>
-            {loading && <ActivityIndicator style={styles.spinner} />}
+            {response.loading && <ActivityIndicator style={styles.spinner} />}
```

Comment:
I guess this should rather be:
```suggestion
            {!newDocuments && <ActivityIndicator style={styles.spinner} />}
```
as we don't need to wait if there are documents in the retry data.